### PR TITLE
rewrote for kafka

### PR DESCRIPTION
**Pull request by xli1996**
# what
This plugin is modified to solve the ce ccs kafka range resolver issue. It will auto set the kafka.version and ce.kafka.version with the correct value in version range, but it will not modify the pom file.

# test
Unit test: basic test to make sure it get ce/ccs correctly.
IT test: pick correct ccs and ce kafka from range [6.0.1-1, 6.0.2-1)
```
6.0.1-12-ce, 6.0.2-1-ccs, 6.0.1-15-ccs, 6.0.1-1000-ce

[INFO] Setting ce.kafka.version property ce.kafka.version=6.0.1-1000-ce
[INFO] Setting kafka.version property kafka.version=6.0.1-15-ccs
```

# unsolved issue
The plugin currently will fail if ce-kafka artifacts can not be fetched.
Currently you can not specify whether you want to resolve for ce or ccs only.
I will fix that on another PR, this PR is provide basic functionality for nano versioning testing.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -256,6 +220,102 @@
...
+                                <phase>verify</phase>
+                                <goals>
+                                    <goal>report</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                      <groupId>org.apache.maven.plugins</groupId>
+                      <artifactId>maven-checkstyle-plugin</artifactId>
```
Comment (ewencp) [inline pom.xml:329]: since  this doesn't have common pom as the parent, these plugins should all specify versions explicitly here

File: pom.xml [hunk 2]
```diff
@@ -256,6 +220,102 @@
...
+                            <execution>
+                                <id>report</id>
+                                <phase>verify</phase>
+                                <goals>
+                                    <goal>report</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
```
Comment (ewencp) [inline pom.xml:327]: Some of these you may want to add incrementally in additional PRs -- things like checkstyle and spotbugs might have fallout in existing code that is unrelated to the goals of this PR

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+ *     -Dresolve.groupId=org.apache.kafka -Dresolve.artifactId=kafka-client "-Dresolve.versionRange=[6.0.0-1, 6.0.1-1]" \
+ *     -Dresolve.CEprint -q
+ * 	   -Dresolve.CCSprint -1
+ * </pre>
+ */
+@Mojo(name = "resolve-kafka-range", defaultPhase = LifecyclePhase.VALIDATE, threadSafe = true, requiresProject = false)
+public class ResolveKafkaVersionRangeMojo extends AbstractMojo {
+	private static final String SNAPSHOT = "SNAPSHOT";
+	private static final String CE = "ce-kafka";
+	private static final String CEVersionString = "-ce";
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: Let's be consistent about the capitalization (which I think you'll need to change anyway as you bring in checkstyle if you are using the same rules as common).

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+	private static final String CCS = "ccs-kafka";
+	private static final String CCSVersionString = "-ccs";
+	private static final Pattern SNAPSHOT_TIMESTAMP = Pattern.compile("^(.*-)?([0-9]{8}\\.[0-9]{6}-[0-9]+)$");
+	private static final String CEKAFKAVERSION = "ce.kafka.version";
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: nit: words should have underscores between them in the variable name

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+	private static final String CCSKAFKAVERSION = "kafka.version";
+
+	/**
+	 * The group id of the artifact to resolve.
+	 */
+	@Parameter(property = "resolve.groupId", required = true)
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: Can we drop the `resolve` prefix from all of these? It doesn't seem necessary? e.g. the other plugins we use don't seem to prefix stuff like this.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+	/**
+	 * Copied from org.eclipse.aether.artifact.AbstractArtifact.isSnapshot(String).
+	 */
+	private static boolean isSnapshot(String version) {
+		return version.endsWith(SNAPSHOT) || SNAPSHOT_TIMESTAMP.matcher(version).matches();
+	}
+
+	private static boolean isCCS(String version) {
+		return version.endsWith(CCSVersionString);
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: Should this be endsWith or matching a regex? Seems like endsWith wouldn't work for SNAPSHOTs.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+						"No matching " + kafkaType + " version found for constraint: '" + constraintText + "'.");
+			}
+
+			return highestVersion;
+		} catch (MojoFailureException e) {
+			throw new MojoExecutionException("", e);
+		}
+	}
+
+	public Version fetchHighestKafkaVersionPublic(String kafkaType, VersionRangeResult result, String constraintText)
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: If this just wraps directly for testing, probably better to just make the original public with a comment `// public for testing`

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+			VersionRangeResult CEResult = repoSystem.resolveVersionRange(repoSession, request);
+			VersionRangeResult CCSResult = repoSystem.resolveVersionRange(repoSession, request);
+			Version highestCEVersion = fetchHighestKafkaVersion(CE, CEResult, constraintText);
+			Version highestCCSVersion = fetchHighestKafkaVersion(CCS, CCSResult, constraintText);
+
+			getLog().info("Highest " + CE + " version: " + highestCEVersion);
+			getLog().info("Highest " + CCS + " version: " + highestCCSVersion);
+
+			if (printCE) {
+				System.out.println(highestCEVersion);
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: This should probably go to the log, not directly to System.out

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+			}
+
+			if (printCCS) {
+				System.out.println(highestCCSVersion);
+			}
+
+			// Set CE property.
+			if (StringUtils.isNotEmpty(CEKAFKAVERSION) && project != null) {
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: Since CEKAFKA_VERSION is a constant that's always non-empty, we can omit that part of the conditional.

File: src/main/java/io/confluent/maven/resolver/ResolveVersionRangeMojo.java [hunk 10]
```diff
@@ -87,6 +87,9 @@
 	@Component
 	private RepositorySystem repoSystem;
 
+	@Parameter(property = "resolve-range.skip", defaultValue = "true")
+	private boolean skip;
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveVersionRangeMojo.java]: instead of making this change can we just delete this class entirely? upstream hasn't changed in years, so I don't think trying to track upstream changes is going to be an issue -- we probably just want to take on all maintenance of this ourselves moving forward.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+ * mvn io.confluent:resolver-maven-plugin:resolve-kafka-range \
+ *     -Dresolve.groupId=org.apache.kafka -Dresolve.artifactId=kafka-client "-Dresolve.versionRange=[6.0.0-1, 6.0.1-1]" \
+ *     -Dresolve.CEprint -q
+ * 	   -Dresolve.CCSprint -1
+ * </pre>
+ */
+@Mojo(name = "resolve-kafka-range", defaultPhase = LifecyclePhase.VALIDATE, threadSafe = true, requiresProject = false)
+public class ResolveKafkaVersionRangeMojo extends AbstractMojo {
+	private static final String SNAPSHOT = "SNAPSHOT";
+	private static final String CE = "ce-kafka";
```
Comment (ewencp) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: I think `CE` and `CCS` might only be getting used in an enum-like way, in which case we should just make them an enum.

File: src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+    private GenericVersionScheme genericVersionScheme = new GenericVersionScheme();
+    private ResolveKafkaVersionRangeMojo resolve = new ResolveKafkaVersionRangeMojo();
+    
+    private static final String ONECCS = "6.0.1-1-ccs";
+    private static final String TWELVECE = "6.0.1-12-ce";
+
+    private final VersionRangeRequest request = new VersionRangeRequest();
+
+    @Test
+    public void testCECCSkafka() throws Exception {
```
Comment (ewencp) [inline src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java:23]: I think this is the critical one where we want the non-lexicographic behavior, but we should get better coverage by validating at least a CE case as well.

File: pom.xml [hunk 13]
```diff
@@ -2,7 +2,7 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>com.subshell.maven</groupId>
+    <groupId>io.confluent</groupId>
     <artifactId>resolver-maven-plugin</artifactId>
     <version>1.0.2-SNAPSHOT</version>
```
Comment (elismaga) [inline pom.xml]: nit: lets reset the version here to 1.0.0 or 0.1.0

File: pom.xml [hunk 14]
```diff
@@ -22,16 +22,6 @@
         </license>
     </licenses>
```
Comment (elismaga) [inline pom.xml:24]: should we remove the url above or change it? also, i think we should remove the license.

File: pom.xml [hunk 15]
```diff
@@ -43,26 +33,10 @@
         <github.global.server>github</github.global.server>
     </properties>
```
Comment (elismaga) [inline pom.xml:31]: is this the version of maven we use? not sure but just checking since it should probably line up.
also, what about the github global server property do we need that?

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+	/**
+	 * The artifact id of the artifact to resolve.
+	 */
+	@Parameter(property = "resolve.artifactId", required = true)
+	private String artifactId;
+
+	/**
+	 * The version range of the artifact to resolve.
+	 */
+	@Parameter(property = "resolve.versionRange", required = true)
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: would it be possible, in another PR, to make this optional and have an optional arg/property you provide called something like version range property. if that is provided it looks in the pom file for that property and uses the value of that as the version range instead?

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+ * GenericVersionScheme</a>
+ * </p>
+ *
+ * <p>
+ * Command-line example:
+ * </p>
+ *
+ * <pre>
+ * mvn io.confluent:resolver-maven-plugin:resolve-kafka-range \
+ *     -Dresolve.groupId=org.apache.kafka -Dresolve.artifactId=kafka-client "-Dresolve.versionRange=[6.0.0-1, 6.0.1-1]" \
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:41]: in the example the versions should have the ce or ccs qualifier.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+				result.setVersions(removeSnapshots(result.getVersions()));
+			}
+			if (kafkaType.equals(CE)) {
+				result.setVersions(includeCE(result.getVersions()));
+				// if ce-kafka can not be fetched
+				if (result.getVersions().isEmpty()) {
+					getLog().info(CE + "can not be fetched");
+				}
+			} else if (kafkaType.equals(CCS)) {
+				result.setVersions(includeCCS(result.getVersions()));
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:193]: why does the CCS steps here not do the same is empty check as the CE stuff above?

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+		try {
+			getLog().debug(kafkaType + " version fetched " + result.getVersions());
+			if (!includeSnapshots) {
+				result.setVersions(removeSnapshots(result.getVersions()));
+			}
+			if (kafkaType.equals(CE)) {
+				result.setVersions(includeCE(result.getVersions()));
+				// if ce-kafka can not be fetched
+				if (result.getVersions().isEmpty()) {
+					getLog().info(CE + "can not be fetched");
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: maybe we should fail in this case? if it can't find a version that satisfies the range then we should probably fail otherwise i am not sure what results running that maven build would give us.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+				}
+			} else if (kafkaType.equals(CCS)) {
+				result.setVersions(includeCCS(result.getVersions()));
+			} else {
+				throw new MojoFailureException("kafka type " + kafkaType + "wrong or not supported");
+			}
+
+			getLog().debug(kafkaType + " Constraint: " + result.getVersionConstraint());
+			getLog().info(kafkaType + " Versions in range: " + result.getVersions());
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: maybe change this to debug since this list will get very long eventually and not sure we need to see the full list of versions in our build logs.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 21]
```diff
@@ -0,0 +1,239 @@
...
+
+			Version highestVersion = result.getHighestVersion();
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:201]: i don't follow whats going on here. after you get the highest version for ce and ccs, then you just get the regular highest version with out the additional logic and return that value? why do we do that?

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 22]
```diff
@@ -0,0 +1,239 @@
...
+ *
+ * <pre>
+ * mvn io.confluent:resolver-maven-plugin:resolve-kafka-range \
+ *     -Dresolve.groupId=org.apache.kafka -Dresolve.artifactId=kafka-client "-Dresolve.versionRange=[6.0.0-1, 6.0.1-1]" \
+ *     -Dresolve.CEprint -q
+ * 	   -Dresolve.CCSprint -1
+ * </pre>
+ */
+@Mojo(name = "resolve-kafka-range", defaultPhase = LifecyclePhase.VALIDATE, threadSafe = true, requiresProject = false)
+public class ResolveKafkaVersionRangeMojo extends AbstractMojo {
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:47]: this is a general comment about the behavior of this plugin/class. i think we can do away with having to specify whether you want to resolve for ce or ccs. the version range or version range property will contain ce or ccs in the versions so we can just determine from that which one the person wants to resolve. also, i don't think we should assume the property the person wants set will adhere to what we have hard coded below. so just in case i think we should take in the name of a property and that is the property we set the version to.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java [hunk 23]
```diff
@@ -0,0 +1,239 @@
...
+			throws MojoExecutionException, MojoFailureException {
+
+		// Workaround for https://issues.apache.org/jira/browse/MNG-3092
+		try {
+			getLog().debug(kafkaType + " version fetched " + result.getVersions());
+			if (!includeSnapshots) {
+				result.setVersions(removeSnapshots(result.getVersions()));
+			}
+			if (kafkaType.equals(CE)) {
+				result.setVersions(includeCE(result.getVersions()));
```
Comment (elismaga) [inline src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:187]: i don't think this works. as we discussed this getting all of the versions that match the range, so ce and ccs and then excluding one or the other, but the sorting that is still being done by maven core and that will still be lexicographical. so 6.0.0-100-ce will be considered a lower version than 6.0.0-2-ce.

**Review by elismaga — CHANGES_REQUESTED** [review]

File: src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java (hunk 12 above)
Comment (elismaga) [reply src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java:23]: Also, I don't think you are really testing to make sure the sort is non-lexicographical. can you make a test with version 6.0.0-100-ce and 6.0.0-2-ce to make sure you get build 100 as the highest. This test is essentially only making sure that when you resolve ccs you don't include ce versions, but it doesn't ensure that you got the correct version/highest version of ccs.

File: src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+
+import static org.junit.Assert.assertEquals;
+import org.eclipse.aether.resolution.VersionRangeRequest;
+import org.eclipse.aether.resolution.VersionRangeResult;
+import org.eclipse.aether.version.Version;
+import org.eclipse.aether.util.version.GenericVersionScheme;
+
+public class KafkaFetchTest {
+    private static final String CCS = "ccs-kafka";
+    private static final String constrainText = "[6.0.1-1, 6.0.2-1)";
```
Comment (elismaga) [inline src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java:13]: why does the version range used for this test not include the -ce or -ccs qualifiers?

**Review by elismaga — CHANGES_REQUESTED** [review]

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 17 above)
Comment (xli1996) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:41]: I figure that we actually do not need to add ce ccs qualifier? I mean the plugin will filter any version without ce or ccs. So kafka can share the same version range with other CP components.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 23 above)
Comment (xli1996) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:187]: At least when i tested locally, the 6.0.0-10000-ce is prior than 6.0.0-12-ce, I might not sure how to mock those maven api call in unit tests, I will see if I can make some unit tests there. The worst case I can always try some IT tests.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 19 above)
Comment (xli1996) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: No, I don't think we should fail it, external people who can not access ce-kafka will fail otherwise.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 18 above)
Comment (xli1996) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:193]: I probably should add that, the CE kafka empty check is only used for cases if people can not access to ce kafka, but if they can not access to ccs either, I guess they will not compile successfully.. but I think it is good to have.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 21 above)
Comment (xli1996) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:201]: The `getHighestVersion` just return the top Version entry at the current Version list, actually the results at first from where I made request is already sorted, all I do essentially is just filter words.

File: src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java (hunk 12 above)
Comment (xli1996) [reply src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java:23]: Actually I did make some other testcases, but then I find the `getHighestVersion` method is just return whatever the top entry in the Version list, which means the actual sorting happen at when you make a VersionRangeRequest. The request seems bit outscope this plugin. I did not find a good way to mock it, maybe I can try some simple IT tests by make some very simple artifacts.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 8 above)
Comment (xli1996) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: This is system.out becuase it will be used as cmd style like 
```
 mvn io.confluent:resolver-maven-plugin:resolve-kafka-range \
 -Dresolve.groupId=org.apache.kafka -Dresolve.artifactId=kafka-client "-Dresolve.versionRange=[6.0.0-1, 6.0.1-1] -D printCCS -q" 
```
which could be useful for CI version update.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 6 above)
Comment (xli1996) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: There is another logic that remove SNAPSHOT first.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 19 above)
Comment (elismaga) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java]: ok, that makes sense when the plugin is running from with in the pom file, but when you are running this from the command line we are going to need different behavior and I think we should fail in that case, or print out None as the result so the script that runs the cmd line can parse the result as None.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 21 above)
Comment (elismaga) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:201]: ok, i get it now. i thought the previous steps were figuring out the latest version but they are just getting a list of versions and this is where you get the latest one from the list. it would probably help to add some more comments around what is going on in this code, but i just didn't follow it for some reason. makes sense now

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 17 above)
Comment (elismaga) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:41]: i don't follow how using a range with out the ce or ccs qualifier would even return you a result that did include artifacts with those qualifiers. either way I think we should still include the qualifiers in the range properties because it would be confusing to people to look in a pom file and see a range like that when they know the artifacts all contain the ce or ccs qualifier. it would end being some mystery to people as to why it worked.

**Review by elismaga — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 15 above)
Comment (xli1996) [reply pom.xml:31]: It is just for whatever maven core version library it used, the build can still be done by any maven version that higher on that.

File: src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java (hunk 22 above)
Comment (elismaga) [reply src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java:47]: discussed offline and decided to address in another PR

**Review by elismaga — APPROVED** [review]
